src/backend/distributed/planner/multi_router_planner.c

```diff
@@ -272,7 +272,7 @@ ErrorIfModifyQueryNotSupported(Query *queryTree)
 				continue;
 			}
 
-			if (!IsA(targetEntry->expr, Const))
+			if (contain_mutable_functions((Node *) targetEntry->expr))
```

Comment:
It's covered by contain_mutable_functions. Onder says they added it to be explicit that naked Var's are allowed.